tests/integration/cattletest/core/test_svc_discovery.py

```diff
@@ -925,6 +876,233 @@ def test_set_service_links(super_client, admin_client,
     assert e.value.error.fieldName == 'serviceId'
 
 
+def _instance_remove(instance, super_client):
+    instance = wait_success(super_client, instance)
+    instance = wait_success(super_client, instance.stop())
+    assert instance.state == 'stopped'
+    instance = wait_success(super_client, instance.remove())
+    assert instance.state == 'removed'
+    return instance
+
+
+def test_destroy_service_instance(super_client,
+                                  admin_client, sim_context, nsp):
+    env = admin_client.create_environment(name=random_str())
+    env = admin_client.wait_success(env)
+    assert env.state == "active"
+
+    image_uuid = sim_context['imageUuid']
+    launch_config = {"imageUuid": image_uuid}
+
+    service = super_client.create_service(name=random_str(),
+                                          environmentId=env.id,
+                                          networkId=nsp.networkId,
+                                          launchConfig=launch_config,
+                                          scale=3)
+    service = super_client.wait_success(service)
+    assert service.state == "inactive"
+
+    # activate service
+    service.activate()
+    service = admin_client.wait_success(service, 120)
+    assert service.state == "active"
+
+    instance1 = _validate_compose_instance_start(super_client, service,
+                                                 env, "1")
+    instance2 = _validate_compose_instance_start(super_client, service,
+                                                 env, "2")
+    instance3 = _validate_compose_instance_start(super_client, service,
+                                                 env, "3")
+
+    # 1. stop and remove the instance2. Validate the mapping still exist
+    instance2 = _instance_remove(instance2, super_client)
+
+    instance_service_map = super_client. \
+        list_serviceExposeMap(serviceId=service.id, instanceId=instance2.id)
+    assert len(instance_service_map) == 1
+    wait_for_condition(
+        super_client, instance_service_map[0], _resource_is_active,
+        lambda x: 'State is: ' + x.state)
+
+    # 2. deactivate the service. the map should still be present
+    service.deactivate()
+    service = admin_client.wait_success(service, 120)
+    assert service.state == "inactive"
+
+    instance_service_map = super_client. \
+        list_serviceExposeMap(serviceId=service.id, instanceId=instance2.id)
+    assert len(instance_service_map) == 1
+    wait_for_condition(
+        super_client, instance_service_map[0], _resource_is_active,
+        lambda x: 'State is: ' + x.state)
+
+    # 3. activate the service. The map should be gone
+    service.activate()
```

Comment:
Do we need to check whether a new instance and mapping is created to preserve the scale?